### ISSUE TITLE
Fabric 1.18 crash on client start

### ISSUE THREAD

**Issue by Shadowrs** [Tfarcenim/FabricFastFurnace#15]
Full log https://mclo.gs/ApWryou
AQM2 modpack with 100+ manually added extra mods - 454 mods total - java 17 - fabric 1.18.2

```

[16:37:43] [ERROR] (FabricLoader) Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.ExceptionInInitializerError
	at Not Enough Crashes deobfuscated stack trace.(1.18.2+build.4) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.8+local.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.polymc.impl.OneSixLauncher.invokeMain(OneSixLauncher.java:104) [NewLaunch.jar:?]
	at org.polymc.impl.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:175) [NewLaunch.jar:?]
	at org.polymc.impl.OneSixLauncher.launch(OneSixLauncher.java:185) [NewLaunch.jar:?]
	at org.polymc.EntryPoint.listen(EntryPoint.java:144) [NewLaunch.jar:?]
	at org.polymc.EntryPoint.main(EntryPoint.java:74) [NewLaunch.jar:?]
Caused by: java.lang.ExceptionInInitializerError
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.1.27.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:33) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:19) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:52) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:86) ~[?:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:85) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:291) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:153) ~[minecraft-1.18.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8+local.jar:?]
	... 11 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_2609 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:419) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8+local.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:186) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:347) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8+local.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks:264) ~[?:?]
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.1.27.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:33) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:19) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:52) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:86) ~[?:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:85) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:291) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:153) ~[minecraft-1.18.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8+local.jar:?]
	... 11 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8+local.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:186) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:347) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8+local.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks:264) ~[?:?]
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.1.27.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:33) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:19) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:52) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:86) ~[?:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:85) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:291) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:153) ~[minecraft-1.18.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8+local.jar:?]
	... 11 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector fast_getFirstMatch(Lnet/minecraft/class_1863;Lnet/minecraft/class_3956;Lnet/minecraft/class_1263;Lnet/minecraft/class_1937;)Ljava/util/Optional; in recipecache.mixins.json:AbstractFurnaceBlockEntityMxn from mod recipecache failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap recipecache-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5-local.jar:0.11.4+mixin.0.8.5-local]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8+local.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:186) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:347) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.8+local.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.8+local.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks:264) ~[?:?]
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.1.27.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:33) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:19) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:52) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:86) ~[?:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:85) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:291) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:153) ~[minecraft-1.18.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.8+local.jar:?]
	... 11 more

```

**Comment by ghost** [comment]
**"recipe cache"** is causing the issue. Try downloading a newer version of recipe cache for 1.18.2.

**Comment by ghost** [comment]
actually, I just saw this closed issue about recipe cache crashing on 1.18.
[**Incompatible with recipecache** #11](https://github.com/Tfarcenim/FabricFastFurnace/issues/11)

**Comment by Tfarcenim** [comment]
shouldn't be an issue on 1.18.2 anymore
